openedx/frontend-lib-content-components
# [BB-7157] Create new editor page for video upload

## Description
This PR adds a new editor page for uploading videos in a course as per the [design doc](https://www.figma.com/file/cS42BX4OUsR8lOw6ppZHoz/Content-Libraries?node-id=3313%3A249596&t=XFyDHuxMZAZAiNZM-0).

## Supporting Information
This PR depends on the changes included in https://github.com/openedx/edx-platform/pull/31869 and https://github.com/openedx/frontend-app-course-authoring/pull/455.

OpenCraft Internal Jira task: [BB-7157](https://tasks.opencraft.com/browse/BB-7157)

## Screenshots
![Screenshot from 2023-04-06 14-26-01](https://user-images.githubusercontent.com/13742492/230464373-0626d681-418d-48f2-a7d7-958e803572a4.png)

![Screenshot from 2023-04-06 14-26-54](https://user-images.githubusercontent.com/13742492/230464338-5e9f7e4a-2f5f-43cd-8842-440510c03895.png)


## Testing Information
1. Ensure your devstack contains the changes from 2 dependent PRs linked above.
2. See https://github.com/openedx/frontend-lib-content-components#how-to-set-up-development-workflow-of-v2-content-editors-in-studio-and-course-authoring-mfe for instructions on how to setup the video editor MFE.
3. Enable Video Uploads page by setting the feature flag `ENABLE_VIDEO_UPLOAD_PIPELINE` to true and setting VideoUploadsEnabledByDefault in the admin panel to enabled.
4. In Studio, select a course and choose `Video Uploads` from the dropdown options under `Content`.
5. You should see the new uploads page load as shown in the screenshot.
6. Drag and drop or click on Browse button to upload required files.
7. Check the console logs for any errors.
